SerialPrograms/Source/PokemonSV/Inference/Overworld/PokemonSV_DirectionDetector.cpp

```diff
@@ -0,0 +1,182 @@
+/*  Direction Detector
+ *
+ *  From: https://github.com/PokemonAutomation/Arduino-Source
+ *
+ */
+
+#include "CommonFramework/GlobalSettingsPanel.h"
+#include "Common/Cpp/Containers/FixedLimitVector.tpp"
+#include "Common/Cpp/Exceptions.h"
+#include "CommonFramework/ImageMatch/WaterfillTemplateMatcher.h"
+#include "CommonFramework/ImageTypes/ImageViewRGB32.h"
+#include "CommonFramework/ImageTools/WaterfillUtilities.h"
+#include "CommonFramework/ImageTools/BinaryImage_FilterRgb32.h"
+#include "CommonFramework/ImageTools/ImageManip.h"
+#include "CommonFramework/VideoPipeline/VideoOverlayScopes.h"
+#include "CommonFramework/Logging/Logger.h"
+#include "CommonFramework/Tools/DebugDumper.h"
+#include "CommonFramework/Tools/VideoResolutionCheck.h"
+#include "NintendoSwitch/Commands/NintendoSwitch_Commands_PushButtons.h"
+#include "Kernels/Waterfill/Kernels_Waterfill_Types.h"
+#include "PokemonSV/Inference/Overworld/PokemonSV_OverworldDetector.h"
+#include "PokemonSV_DirectionDetector.h"
+#include <cmath>
+#include <iostream>
+#ifndef M_PI
```

Comment:
why is this needed ? It should be included with cmath ? (if it's not, try defining _USE_MATH_DEFINES)

Reply:
I tried using _USE_MATH_DEFINES instead of manually defining M_PI. It seems to be causing the Windows 5.12.12 build to fail.

```D:\a\Arduino-Source\Arduino-Source\Arduino-Source\SerialPrograms\Source\PokemonSV\Inference\Overworld\PokemonSV_DirectionDetector.cpp(129,52): error C2065: 'M_PI': undeclared identifier [D:\a\Arduino-Source\Arduino-Source\Arduino-Source\SerialPrograms\bin\SerialPrograms.vcxproj]```

Any ideas on how to fix this?